core/src/main/java/io/kestra/core/models/executions/Execution.java

```diff
@@ -136,7 +136,7 @@ public static Execution newExecution(final Flow flow, final List<Label> labels)
     }
 
     public List<Label> getLabels() {
-        return Optional.ofNullable(this.labels).orElse(new ArrayList<>());
+        return Label.deduplicate(this.labels);
```

Comment:
I think it's better to deduplicate before setting the labels inside the Execution than when getting the labels from the execution.

Reply:
And we have an helper to avoid NPE on list: `ListUtils.emptyOnNull(someList)`.